src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -0,0 +1,408 @@
+                        return GetAbstractMemberSymbol(memberResult.Member);
+                    }
+                    else
+                    {
+                        return memberResult.Member;
+                    }
+                });
+            var options = new CodeGenerationOptions(reuseSyntax: true, generateMethodBodies: false);
+            var newDestination = codeGenerationService.AddMembers(destinationSyntaxNode, pullUpMembersSymbols, options: options);
+
+            // Remove some original members since we are pulling members into class.
+            // Note: If user chooses to make the member abstract, then the original member won't be touched,
+            // It will just pull abstract declaration up to destination.
+            // But if the member is abstract itself, it will still be removed.
+            foreach (var analysisResult in result.MemberAnalysisResults)
+            {
+                if (!analysisResult.MakeDeclarationAtDestinationAbstract)
+                {
+                    foreach (var syntax in syntaxMap[analysisResult.Member])
+                    {
+                        var originalMemberEditor = editorMap[syntax.SyntaxTree];
+                        originalMemberEditor.RemoveNode(originalMemberEditor.Generator.GetDeclaration(syntax));
+                    }
+                }
+            }
+
+            // Change the destination to abstract class if needed.
+            var destinationEditor = editorMap[destinationSyntaxNode.SyntaxTree];
+            if (!result.Destination.IsAbstract &&
+                result.MemberAnalysisResults.Any(analysis => analysis.Member.IsAbstract || analysis.MakeDeclarationAtDestinationAbstract))
+            {
+                var modifiers = DeclarationModifiers.From(result.Destination).WithIsAbstract(true);
+                newDestination = destinationEditor.Generator.WithModifiers(newDestination, modifiers);
+            }
+
+            destinationEditor.ReplaceNode(destinationSyntaxNode, (syntaxNode, generator) => newDestination);
+            return solutionEditor.GetChangedSolution();
+        }
+
+        private ISymbol GetAbstractMemberSymbol(ISymbol member)
+        {
+            if (member.IsAbstract)
+            {
+                return member;
+            }
+
+            var modifier = DeclarationModifiers.From(member).WithIsAbstract(true);
+            if (member is IMethodSymbol methodSymbol)
+            {
+                return CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, modifiers: modifier);
+            }
+            else if (member is IPropertySymbol propertySymbol)
+            {
+                return CodeGenerationSymbolFactory.CreatePropertySymbol(propertySymbol, modifiers: modifier);
+            }
+            else if (member is IEventSymbol eventSymbol)
+            {
+                return CodeGenerationSymbolFactory.CreateEventSymbol(eventSymbol, modifiers: modifier);
+            }
+            else
```

Comment:
what prevented you from calling into this with a field?

Reply:
Emmm, I think I should add a double check in the caller of this method. Since currently it rely on the UI to do it. (yeah, since all fields don't have an 'Make Abstract' checkbox, so the input from the UI will not be able to do so)